### ISSUE TITLE
honeycombio_burn_alert > dynamic `recipient` "Value Conversion Error" 

### ISSUE THREAD

**Issue by bgroff-trdrev** [honeycombio/terraform-provider-honeycombio#434]
Terraform Validate recieves the following Value Conversion Error whenever trying to dynamic block the burn alert `recipients`.

```
╷
│ Error: Value Conversion Error
│ 
│   with honeycombio_burn_alert.burn_alert,
│   on slos.tf line 29, in resource "honeycombio_burn_alert" "burn_alert":
│   29: resource "honeycombio_burn_alert" "burn_alert" {
│ 
│ An unexpected error was encountered trying to build a value. This is always an error in the provider. Please report the following to the provider developer:
│ 
│ Received unknown value, however the target type cannot handle unknown values. Use the corresponding `types` package type or a custom type that handles unknown values.
│ 
│ Path: recipient
│ Target Type: []models.NotificationRecipientModel
│ Suggested Type: basetypes.SetValue
```
**Versions**
Affected honeycombio provider versions: **0.20.0 - 0.21.0**.
appears to work when downgrading to 0.19.0
```tf
terraform {
  required_version = "~> 1.5.7"
  required_providers {
    honeycombio = {
      source  = "honeycombio/honeycombio"
      version = "~> 0.21.0"
    }
  }
}

```

**Steps to reproduce**

```tf
variable "alert_recipients" {
  type = list(object({
    type   = string
    target = string
  }))
  description = "recipients to use for trigger and slo alerting"

  validation {
    condition     = alltrue([for rec in var.alert_recipients : contains(["slack", "email"], rec.type)])
    error_message = "Recipient types must be one of: [slack, email]"
  }
}

variable "slos" {
  description = "A list of SLO resources"
  type = list(object({
    name        = string
    description = optional(string)
    derived_column = object({
      alias      = string
      expression = string
    })
    target_percentage = optional(number)
    time_period       = optional(number)
    burn_alerts = optional(list(object({
      alert_type         = optional(string)
      exhaustion_minutes = optional(number)
    })))
  }))
  default = []
}

resource "honeycombio_derived_column" "slo_derived_column" {
  for_each   = { for slo in var.slos : slo.name => slo }
  dataset    = var.dataset
  alias      = "${each.key}_${each.value.derived_column.alias}"
  expression = each.value.derived_column.expression
}

resource "honeycombio_slo" "slo" {
  for_each          = { for slo in var.slos : slo.name => slo }
  name              = each.value.name
  dataset           = var.dataset
  description       = "${try(each.value.description, "")}"
  sli               = honeycombio_derived_column.slo_derived_column[each.key].alias
  target_percentage = try(each.value.target_percentage, null)
  time_period       = try(each.value.time_period, null)
}

locals {
  flat_burn_alerts = flatten([
    for idx, slo in var.slos : [
      for burn in slo.burn_alerts : {
        slo_name   = slo.name
        burn_alert = burn
      }
    ]
  ])
}

resource "honeycombio_burn_alert" "burn_alert" {
  for_each           = local.flat_burn_alerts
  dataset            = var.dataset
  slo_id             = honeycombio_slo.slo[each.value.slo_name].id
  exhaustion_minutes = try(each.value.burn_alert.exhaustion_minutes, null)
  alert_type         = try(each.value.burn_alert.alert_type, null)

  dynamic "recipient" {
    for_each = var.alert_recipients

    content {
      type   = recipient.value.type
      target = recipient.value.target
    }
  }
}

```

1. Setup above sample in `main.tf`. Also setup prior snippet for providers.
2. `terraform init`
3. `terraform validate` and see error

**Additional context**
using `tfenv` to manage versions, I tested from the latest version down to 0.19.0 before the issue was resolved. So somewhere between 0.19.0 and 0.20.0 the issue was created. Note that 0.19.0 does not support the `alert_type` field so I had to comment that out prior to validating that version of the provider.

**Comment by jharley** [comment]
Thanks for the detailed bug report, @bgroff-trdrev.

We'll look into it shortly.

**Comment by zacMode** [comment]
As another datapoint - we were facing the same issue using v0.11 (a very old 😬 version) of the honeycomb provider. 

Upgrading to 0.21.0 caused the same issue, but using 0.19.0 fixed the issue.

**Comment by jharley** [comment]
More information here: https://discuss.hashicorp.com/t/resource-createrequest-plan-get-throws-error-on-slice-of-struct-in-resource-model/56887

**Comment by jharley** [comment]
@bgroff-trdrev a fix for this just went out in v0.22.0
